services/brig/src/Brig/API/Public.hs

```diff
@@ -868,51 +783,36 @@ getMultiUserPrekeyBundleH zusr qualUserClients = do
   API.claimMultiPrekeyBundles (ProtectedUser zusr) qualUserClients !>> clientError
 
-addClientH :: JsonRequest Public.NewClient ::: UserId ::: ConnId ::: Maybe IpAddr ::: JSON -> Handler Response
-addClientH (req ::: usr ::: con ::: ip ::: _) = do
-  new <- parseJsonBody req
-  clt <- addClient new usr con ip
-  let loc = toByteString' $ Public.clientId clt
-  pure . setStatus status201 . addHeader "Location" loc . json $ clt
-
-addClient :: Public.NewClient -> UserId -> ConnId -> Maybe IpAddr -> Handler Public.Client
-addClient new usr con ip = do
+addClient :: UserId -> ConnId -> Maybe IpAddr -> Public.NewClient -> Handler BrigAPI.NewClientResponse
+addClient usr con ip new = do
   -- Users can't add legal hold clients
   when (Public.newClientType new == Public.LegalHoldClientType) $
     throwE (clientError ClientLegalHoldCannotBeAdded)
-  API.addClient usr (Just con) (ipAddr <$> ip) new !>> clientError
-
-rmClientH :: JsonRequest Public.RmClient ::: UserId ::: ConnId ::: ClientId ::: JSON -> Handler Response
-rmClientH (req ::: usr ::: con ::: clt ::: _) = do
-  body <- parseJsonBody req
-  empty <$ rmClient body usr con clt
+  clientResponse <$> API.addClient usr (Just con) (ipAddr <$> ip) new !>> clientError
+  where
+    clientResponse :: Public.Client -> BrigAPI.NewClientResponse
+    clientResponse client = Servant.addHeader (Public.clientId client) client
 
-rmClient :: Public.RmClient -> UserId -> ConnId -> ClientId -> Handler ()
-rmClient body usr con clt =
+deleteClient :: UserId -> ConnId -> ClientId -> Public.RmClient -> Handler (EmptyResult 200)
+deleteClient usr con clt body = do
   API.rmClient usr con clt (Public.rmPassword body) !>> clientError
+  pure EmptyResult
 
-updateClientH :: JsonRequest Public.UpdateClient ::: UserId ::: ClientId ::: JSON -> Handler Response
-updateClientH (req ::: usr ::: clt ::: _) = do
-  body <- parseJsonBody req
-  empty <$ updateClient body usr clt
-
-updateClient :: Public.UpdateClient -> UserId -> ClientId -> Handler ()
-updateClient body usr clt =
-  API.updateClient usr clt body !>> clientError
-
-listClientsH :: UserId ::: JSON -> Handler Response
-listClientsH (zusr ::: _) =
-  json <$> listClients zusr
+updateClient :: UserId -> ClientId -> Public.UpdateClient -> Handler (EmptyResult 200)
+updateClient usr clt upd = do
+  API.updateClient usr clt upd !>> clientError
+  pure EmptyResult
 
 listClients :: UserId -> Handler [Public.Client]
 listClients zusr =
   lift $ API.lookupLocalClients zusr
 
-getClientH :: UserId ::: ClientId ::: JSON -> Handler Response
-getClientH (zusr ::: clt ::: _) =
-  getClient zusr clt <&> \case
-    Just c -> json c
-    Nothing -> setStatus status404 empty
+getClient :: UserId -> ClientId -> Handler (Union BrigAPI.GetClientResponse)
+getClient zusr clientId = do
+  mc <- lift $ API.lookupLocalClient zusr clientId
+  case mc of
+    Nothing -> Servant.respond ErrorDescription.clientNotFound
+    Just c -> Servant.respond (WithStatus @200 c)
```

Comment:
Does this mean we used to return empty body earlier and now we return the error description? I hope this doesn't break anything :crossed_fingers:

Reply:
Yeah. I'm not sure how else to handle this, since `Servant` really does not allow an empty body unless the list of content types is empty, and we cannot have an empty list of content types, because we return a non-empty body in one of the cases.

Reply:
Hmm, I think it is a mistake to set and empty body in this case anyway and the likelihood of this breaking anything is small, so maybe it is ok. Unfortunate that we'll have to wait for a merge to know if this actually breaks anything. Maybe we can ask client devs, if they think this will break stuff?